tests/functional/sync_v2_sign_up.js

```diff
@@ -0,0 +1,142 @@
+
+define([
+  'intern',
+  'intern!object',
+  'intern/chai!assert',
+  'require',
+  'intern/node_modules/dojo/node!xmlhttprequest',
+  'app/bower_components/fxa-js-client/fxa-client',
+  'tests/lib/helpers',
+  'tests/functional/lib/helpers'
+], function (intern, registerSuite, assert, require, nodeXMLHttpRequest,
+        FxaClient, TestHelpers, FunctionalHelpers) {
+  'use strict';
+
+  var config = intern.config;
+  var PAGE_URL = config.fxaContentRoot + 'signup?context=iframe&service=sync';
+
+  var SIGNIN_URL = config.fxaContentRoot + 'signin';
+
+  var AUTH_SERVER_ROOT = config.fxaAuthRoot;
+  var TOO_YOUNG_YEAR = new Date().getFullYear() - 13;
+  var OLD_ENOUGH_YEAR = TOO_YOUNG_YEAR - 1;
+
+  var client;
+  var email;
+  var PASSWORD = '12345678';
+
+  var listenForFxaCommands = FunctionalHelpers.listenForWebChannelMessage;
+  var respondToWebChannel = function (context, expectedCommand, response) {
+    return function () {
+      return context.get('remote')
+        .execute(function (expectedCommand, response) {
+          /* global addEventListener, removeEventListener, CustomEvent, dispatchEvent */
+          addEventListener('WebChannelMessageToChrome', function listener(e) {
+            removeEventListener('WebChannelMessageToChrome', listener);
+
+            var command = e.detail.message.command;
+            var messageId = e.detail.message.messageId;
+
+            if (command === expectedCommand) {
+              var event = new CustomEvent('WebChannelMessageToContent', {
+                detail: {
+                  id: 'account_updates',
+                  message: {
+                    messageId: messageId,
+                    command: command,
+                    data: response
+                  }
+                }
+              });
+
+              dispatchEvent(event);
+            }
+          });
+        }, [ expectedCommand, response ]);
+    };
+  };
+
+  registerSuite({
+    name: 'Firefox Desktop Sync v2 sign_up',
+
+    beforeEach: function () {
+      email = TestHelpers.createEmail();
+
+      client = new FxaClient(AUTH_SERVER_ROOT, {
+        xhr: nodeXMLHttpRequest.XMLHttpRequest
+      });
+
+      return FunctionalHelpers.clearBrowserState(this);
+    },
+
+    teardown: function () {
+      var self = this;
+
+      return FunctionalHelpers.clearBrowserState(this)
+        .then(function () {
+          // ensure the next test suite (bounced_email) loads a fresh
+          // signup page. If a fresh signup page is not forced, the
+          // bounced_email tests try to sign up using the Sync broker,
+          // resulting in a channel timeout.
+          self.get('remote')
+            .get(require.toUrl(SIGNIN_URL))
+
+            .findByCssSelector('#fxa-signin-header')
+            .end();
+        });
+    },
+
+    'sign up, verify same browser': function () {
+      var self = this;
+
+      return FunctionalHelpers.openPage(this, PAGE_URL)
+        .execute(listenForFxaCommands)
```

Comment:
Seehttps://github.com/mozilla/fxa-content-server/pull/2237/files#r30061298
`listenForFxa...` does  document.body.appendChild(element); this could cause issues on slow connections